internal/policies/mount/mount.go

```diff
@@ -164,12 +196,174 @@ func (m *Manager) applyUserMountsPolicy(ctx context.Context, username string, en
 	return nil
 }
 
+func (m *Manager) applySystemMountsPolicy(ctx context.Context, machineName string, entry entry.Entry) (err error) {
+	decorate.OnError(&err, "error when applying mount policy to machine %q", machineName)
+
+	log.Debugf(ctx, "Applying mount policy to machine %q", machineName)
+
+	parsedValues, err := parseEntryValues(entry)
+	if err != nil {
+		return err
+	}
+	newUnits := createUnits(parsedValues)
+
+	// Marks shares to write as new units and removes from map units that shouldn't change
+	needsReload := false
+	var unitsToEnable []string
+
+	prevUnits := m.currentSystemMountUnits()
+
+	// Removes from the map all the units that are supposed to be written or updated.
+	for name := range newUnits {
+		delete(prevUnits, name)
+	}
+
+	var unitsToClean []string
+	for name := range prevUnits {
+		unitsToClean = append(unitsToClean, name)
+	}
+
+	if err := m.cleanupMountUnits(ctx, unitsToClean); err != nil {
+		return err
+	}
+
+	for name, content := range newUnits {
+		written, err := writeIfChanged(filepath.Join(m.systemUnitDir, name), content)
+		if err != nil {
+			return err
+		}
+		if written {
+			unitsToEnable = append(unitsToEnable, name)
+		}
+		needsReload = needsReload || written
+	}
+
+	if !needsReload {
+		return nil
+	}
+
+	// Trigger a daemon reload
+	if err := m.execSystemCtlCmd(ctx, "daemon-reload"); err != nil {
+		return fmt.Errorf("failed to reload systemctl: %w", err)
+	}
+
+	// Enables and starts new units.
+	for _, name := range unitsToEnable {
+		if err := m.execSystemCtlCmd(ctx, "enable", name); err != nil {
+			return fmt.Errorf("failed to enable unit %q: %w", name, err)
+		}
+		if err := m.execSystemCtlCmd(ctx, "start", name); err != nil {
+			log.Warningf(ctx, "failed to start unit %q: %v", name, err)
+		}
```

Comment:
Yeah, `systemd enable` is almost the same thing as just calling `ln`. I feel like the `systemd-analyze verify` won't help much, as the man page says:
```bash
The following errors are currently detected:
       •   unknown sections and directives,
       •   missing dependencies which are required to start the given unit,
       •   man pages listed in Documentation= which are not found in the system,
       •   commands listed in ExecStart= and similar which are not found in the system
```

But now that you commented on it, I feel like this logic might be wrong all together. Since we defined that enabling units is a point of failure and starting them isn't, I should probably try to enable all units before starting them, sounds more correct.

Reply:
I agree that enable has a very low chance of failure, but it can still fail... Since we are very much in control of the generated units and `systemd-analyze` does only a very superficial check, it's better to just leave it as is (this way, we don't rely on another command to apply the policy).
Also, this sytemctl call is a temporary thing as well... We are thinking about replacing this calls with dbus communication.

Reply:
TBH, I’m with Denis on this, we are using a template and generates from it. If the generated file is wrong, this could be part of a test (and running analyze as part of the test for a particular protocol).